src/main/java/org/edumips64/ui/swing/DineroFrontend.java

```diff
@@ -243,15 +350,165 @@ public void keyReleased(KeyEvent e) {
     cp.add(cmdLine);
     cp.add(Box.createRigidArea(vSpace));
 
+    //Box created for adding Cache Panel components
+		cachePanel = Box.createVerticalBox();
+		cachePanel.add(panelL1 = new DineroSingleCachePanel(argType, 1));
+		cp.add(cachePanel);
+    cp.add(Box.createRigidArea(vSpace));
+    
     result = new JTextArea();
     result.setBorder(BorderFactory.createTitledBorder("Messages"));
     result.setEditable(false);
     result.setFont(new Font("Monospaced", Font.PLAIN, config.getInt(ConfigKey.UI_FONT_SIZE)));
 
+    cp.add(configure);
+		cp.add(Box.createRigidArea(vSpace));
     cp.add(execute);
     cp.add(Box.createRigidArea(vSpace));
     cp.add(new JScrollPane(result));
 
-    setSize(850, 500);
+    //Vertical and horizontal scroll added to the Frame container
+		scrollPane = new JScrollPane(cp, JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+		setContentPane(scrollPane);
+
+		//Resized the frame container to have better UX with the new panels
+		setSize(900, 800);
   }
 }
+
+/** Panel with all the necessary controls to modify the options of a Cache.
+ */
+class DineroSingleCachePanel extends JPanel {
+	private DineroCacheOptions dco;
+	private JComboBox size, sizeUnit, bsize, bsizeUnit, level, type;
```

Comment:
Same here, please use `JComboBox<String>`.

Also, looks like `lwvel` and `type` are not used, please remove them.

Reply:
The changes have been made. Thank you.